File: z-clients/javascript/tests/integration.test.mjs
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { Diom } from "../dist/index.mjs";

const TOKEN = process.env.DIOM_TOKEN;
const SERVER_URL = process.env.DIOM_SERVER_URL;

if (!TOKEN || !SERVER_URL) {
  throw new Error("DIOM_TOKEN and DIOM_SERVER_URL environment variables must be set");
}

function makeClient() {
  return new Diom(TOKEN, { serverUrl: SERVER_URL });
}

function toBytes(str) {
  return Array.from(Buffer.from(str));
}

describe("SDK Integration Tests", () => {
  it("test_health_ping", async () => {
    const client = makeClient();
    const resp = await client.health.ping();
    assert.ok(resp);
  });

  it("test_kv_set_get_delete", async () => {
    const client = makeClient();
    const key = "js-integration-kv-key";
    const value = toBytes("js-integration-kv-value");

    // Set
    const setResp = await client.kv.set(key, { value });
    assert.strictEqual(setResp.success, true);

    // Get
    const getResp = await client.kv.get(key, {});
    assert.deepStrictEqual(Array.from(getResp.value), value);

    // Delete
    const delResp = await client.kv.delete(key, {});
    assert.strictEqual(delResp.success, true);

    // Verify deleted
    const getResp2 = await client.kv.get(key, {});
    assert.strictEqual(getResp2.value, null);
  });

  it("test_cache_set_get_delete", async () => {
    const client = makeClient();
    const key = "js-integration-cache-key";
    const value = toBytes("js-integration-cache-value");

    // Set
    await client.cache.set(key, { value, ttlMs: 60000 });

    // Get
    const getResp = await client.cache.get(key, {});
    assert.deepStrictEqual(Array.from(getResp.value), value);

    // Delete
    const delResp = await client.cache.delete(key, {});
    assert.strictEqual(delResp.success, true);

    // Verify deleted
    const getResp2 = await client.cache.get(key, {});
    assert.strictEqual(getResp2.value, null);
  });
});

File: z-clients/java/src/test/java/com/svix/diom/IntegrationTest.java
package com.svix.diom;

import static org.junit.jupiter.api.Assertions.*;

import com.svix.diom.models.*;
import java.time.Duration;
import java.util.Collections;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

@Tag("integration")
class IntegrationTest {

    private static final String TOKEN = requireEnv("DIOM_TOKEN");
    private static final String SERVER_URL = requireEnv("DIOM_SERVER_URL");

    private static String requireEnv(String name) {
        String value = System.getenv(name);
        if (value == null || value.isEmpty()) {
            throw new IllegalStateException(name + " must be set");
        }
        return value;
    }

    private static Diom client;

    @BeforeAll
    static void setUp() {
        DiomOptions opts = new DiomOptions();
        opts.setServerUrl(SERVER_URL);
        client = new Diom(TOKEN, opts);
    }

    @Test
    @Tag("integration")
    void testHealthPing() throws Exception {
        PingOut resp = client.health().ping();
        assertNotNull(resp);
    }

    @Test
    @Tag("integration")
    void testKvSetGetDelete() throws Exception {
        String key = "java-integration-kv-key";
        byte[] value = "java-integration-kv-value".getBytes();

        // Set
        KvSetOut setResp = client.kv().set(key, value);

        // Get
        KvGetOut getResp = client.kv().get(key);
        assertArrayEquals(value, getResp.getValue());

        // Delete
        KvDeleteOut delResp = client.kv().delete(key);
        assertTrue(delResp.getSuccess());

        // Verify deleted
        KvGetOut getResp2 = client.kv().get(key);
        assertNull(getResp2.getValue());
    }

    @Test
    @Tag("integration")
    void testMsgsQueueSubresourceChaining() throws Exception {
        String namespace = "java-integration-ns";
        String topic = "java-integration-topic";
        String consumerGroup = "java-integration-cg";

        client.msgs().namespace().configure(namespace);
        client.msgs().publish(topic, new MsgPublishIn()
            .namespace(namespace)
            .msgs(Collections.singletonList(new MsgIn().value("hello".getBytes()))));

        MsgQueueReceiveOut received = client.msgs().queue().receive(
            topic, consumerGroup, new MsgQueueReceiveIn().namespace(namespace));
        assertNotNull(received);
        assertFalse(received.getMsgs().isEmpty());

        String msgId = received.getMsgs().get(0).getMsgId();
        MsgQueueAckOut ackResp = client.msgs().queue().ack(
            topic, consumerGroup,
            new MsgQueueAckIn()
                .namespace(namespace)
                .msgIds(Collections.singletonList(msgId)));
        assertNotNull(ackResp);
    }

    @Test
    @Tag("integration")
    void testCacheSetGetDelete() throws Exception {
        String key = "java-integration-cache-key";
        byte[] value = "java-integration-cache-value".getBytes();

        // Set
        client.cache().set(key, value, new CacheSetIn().ttl(Duration.ofMillis(60000)));

        // Get
        CacheGetOut getResp = client.cache().get(key);
        assertArrayEquals(value, getResp.getValue());

        // Delete
        CacheDeleteOut delResp = client.cache().delete(key);
        assertTrue(delResp.getSuccess());

        // Verify deleted
        CacheGetOut getResp2 = client.cache().get(key);
        assertNull(getResp2.getValue());
    }
}
